src/approaches/nsrt_learning_approach.py

```diff
@@ -85,3 +89,61 @@ def load(self, online_learning_cycle: Optional[int]) -> None:
         # Seed the option parameter spaces after loading.
         for nsrt in self._nsrts:
             nsrt.option.params_space.seed(CFG.seed)
+
+    def _compute_sidelining_objective_value(
+            self, trajectories: List[LowLevelTrajectory]) -> None:
+        """Compute the value of the objective function that sidelining is
+        trying to approximately optimize.
+
+        Store this into self._metrics.
+        """
+        logging.info("Computing sidelining objective value...")
+        start_time = time.time()
+        preds = self._get_current_predicates()
+        strips_ops = [nsrt.op for nsrt in self._nsrts]
+        option_specs = [(nsrt.option, list(nsrt.option_vars))
+                        for nsrt in self._nsrts]
+        # Calculate first term in objective. This is a sum over the training
+        # tasks of the number of possible task plans up to the demo length.
+        num_plans_up_to_n = 0
+        for segment_traj, ll_traj in zip(self._segmented_trajs, trajectories):
+            if not ll_traj.is_demo:
+                continue
+            task = self._train_tasks[ll_traj.train_task_idx]
+            init_atoms = utils.abstract(task.init, preds)
+            objects = set(task.init)
+            ground_nsrts, reachable_atoms = task_plan_grounding(
+                init_atoms,
+                objects,
+                strips_ops,
+                option_specs,
+                allow_noops=True)
+            heuristic = utils.create_task_planning_heuristic(
+                CFG.sesame_task_planning_heuristic, init_atoms, task.goal,
+                ground_nsrts, preds, objects)
+            for skeleton, _, _ in task_plan(init_atoms,
+                                            task.goal,
+                                            ground_nsrts,
+                                            reachable_atoms,
+                                            heuristic,
+                                            CFG.seed,
+                                            timeout=10000000,
```

Comment:
is this going to potentially kill us if we use this with baselines that have argument explosions?